### PR TITLE
Add 'cm_tools_get_block' function.

### PR DESCRIPTION
**Pull request by stevetweeddale**
Credit to James Williams for this one. We like?

### CODE REVIEW

File: includes/block.inc [hunk 1]
```diff
@@ -49,3 +49,20 @@ function cm_tools_block_position($module, $delta, $block = array()) {
...
 }
+
+/**
+ * Get a render array for a block, no questions asked.
+ *
+ * @param $module
+ *   The module that provides the block.
+ * @param $delta
+ *   The block delta.
+ */
```
Comment (darthsteven) [inline includes/block.inc:64]: You need to define the return value of the function using a @return style block.

**Comment by jamsilver** [comment]
Does that approach:
1. Go through block_view_alter?
2. Correctly render contextual links?
3. Honour changes to the block title made through the UI?

File: includes/block.inc [hunk 2]
```diff
@@ -49,3 +49,20 @@ function cm_tools_block_position($module, $delta, $block = array()) {
     )))
     ->execute();
 }
+
+/**
+ * Get a render array for a block, no questions asked.
```
Comment (darthsteven) [inline includes/block.inc]: Can we change this too: 'Retrieve the render array representation for the given block.' please?

**Comment by anotherjames** [comment]
@jamsilver _block_render_blocks() does send it through drupal_alter(). _block_get_renderable_array() adds the contextual links.

**Comment by anotherjames** [comment]
@jamsilver Editing comments is bad. But, yes, config made to the block in the UI is pulled through, including the title. block_load() pulls the data through, _block_render_blocks() uses it.

**Pull request merged**